From 39382419f7e873189d69990cc45e2f6c4c47ba2c Mon Sep 17 00:00:00 2001
From: Kyle Taylor <kyletaylored@gmail.com>
Date: Mon, 28 Jul 2014 13:52:04 -0500
Subject: [PATCH] Add wrapper/class options to Icon Block

---
 modules/icon_block/icon_block.module | 38 ++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/modules/icon_block/icon_block.module b/modules/icon_block/icon_block.module
index 7a59dcd..23e705e 100644
--- a/modules/icon_block/icon_block.module
+++ b/modules/icon_block/icon_block.module
@@ -25,6 +25,8 @@ function icon_block_defaults() {
     'icon' => '',
     'position' => 'title_before',
     'bundle' => '',
+    'wrapper' => 'none',
+    'wrapper_class' => '',
   );
 }
 
@@ -38,6 +40,7 @@ function icon_block_preprocess_block(&$variables) {
   $settings = array_merge(icon_block_defaults(), isset($block->icon) ? (array) unserialize($block->icon) : array());
   if (!empty($settings['bundle']) && !empty($settings['icon'])) {
     if ($icon = theme('icon', array('bundle' => $settings['bundle'], 'icon' => $settings['icon']))) {
+      $icon = _icon_block_class_wrapper($icon, $settings['wrapper'], $settings['wrapper_class']);
       $variables['classes_array'][] = 'has-icon';
       switch ($settings['position']) {
         case 'title_before':
@@ -100,6 +103,29 @@ function icon_block_form_alter(&$form, &$form_state, $form_id) {
       '#default_value' => $settings['position'],
       '#states' => $icon_state,
     );
+    $form['settings']['icon_selector']['wrapper'] = array(
+      '#access' => $access,
+      '#type' => 'select',
+      '#title' => t('Wrapper'),
+      '#options' => array(
+        'none' => t('None'),
+        'h2' => t('H2'),
+        'h3' => t('H3'),
+        'h4' => t('H4'),
+        'div' => t('DIV'),
+        'span' => t('SPAN'),
+      ),
+      '#default_value' => $settings['wrapper'],
+      //'#states' => $icon_state,
+    );
+    $form['settings']['icon_selector']['wrapper_class'] = array(
+      '#access' => $access,
+      '#type' => 'textfield',
+      '#title' => t('Wrapper Class'),
+      '#default_value' => $settings['wrapper_class'],
+      '#size' => 60, 
+      '#maxlength' => 128, 
+    );
     if ($access) {
       $form['#submit'][] = 'icon_block_form_submit';
     }
@@ -119,3 +145,15 @@ function icon_block_form_submit($form, &$form_state) {
     }
   }
 }
+
+/*
+ * Helper function for wrapping icons.
+ */
+function _icon_block_class_wrapper($string, $element = '', $class = '') {
+  $cl = !empty($class) ? $class : '';
+  if (empty($element) || $element == 'none') {
+    return $string;
+  } else {
+    return '<'.$element.' class="'.$class.'">'.$string.'</'.$element.'>';
+  }
+}
-- 
1.8.5.2 (Apple Git-48)

